### PR TITLE
Add PHP Syntax Error sniff to WordPressVIPMinimum

### LINKED ISSUE #289: Syntax Error Sniff
## What problem would the enhancement address for VIP?

The `WordPress-Extra` sniffs include syntax error detection, but since we don't extend that (just cherry pick out various WP sniffs via `ruleset.xml`, we are missing syntax errors.

test.php
```
<?php

thisisasyntaxerror
```

Running the ruleset produces no errors though:

```
$ phpcs --standard=WordPressVIPMinimum test.php
$
```

## Describe the solution you'd like

Syntax errors are caught by the `WordPressVIPMinimum` and `WordPress-VIP-Go` rulesets.

## What code should be reported as a violation?

Any PHP syntax error.

## What could should *not* be reported as a violation?

Valid PHP code.

## Additional context

We're missing this:

```
<rule ref="Generic.PHP.Syntax"/>
```

Right now we're picking certain rules out of the WordPress rulesets - would it make more sense to include the full ruleset and then exclude rules we don't want?

### PR DESCRIPTION
**Pull request by nickdaugherty**
Has been inadvertently omitted until now, but it's arguably the most important sniff to have.

Fixes #289

### CODE REVIEW

File: WordPressVIPMinimum/ruleset.xml [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 	<autoload>./PHPCSCompatibility.php</autoload>
 
+	<rule ref="Generic.PHP.Syntax" />
```
Comment (GaryJones) [inline WordPressVIPMinimum/ruleset.xml]: There should be no space before the `/>`. This is XML, not XHTML.

**Comment by GaryJones** [comment]
Nothing that this change will affect both `WordPressVIPMinimum` and `WordPress-VIP-Go` rulesets (since the latter inherits the former).

**Comment by gudmdharalds** [comment]
> @gudmdharalds Since the vip-ci-bot already does php -l, should anything change there once this commit is used by the bot, to avoid duplicate reporting?

Yep, we will have to skip this sniff. There is also another sniff to ignore, so I will add an option to `vip-go-ci` to make sure we can skip custom sniffs.

**Comment by gudmdharalds** [comment]
> @gudmdharalds Since the vip-ci-bot already does php -l, should anything change there once this commit is used by the bot, to avoid duplicate reporting?

Yep, we will have to skip this sniff. There is also another sniff to ignore, so I will add an option to `vip-go-ci` to make sure we can skip custom sniffs.

**Comment by gudmdharalds** [comment]
Yep, we will have to skip this sniff. There is also another sniff to ignore, so I will add an option to `vip-go-ci` to make sure we can skip custom sniffs.

**Comment by gudmdharalds** [comment]
> @gudmdharalds Since the vip-ci-bot already does php -l, should anything change there once this commit is used by the bot, to avoid duplicate reporting?

Yep, we will have to skip this sniff. There is also another sniff to ignore, so I will add an option to `vip-go-ci` to make sure we can skip custom sniffs.

**Comment by gudmdharalds** [comment]
> @gudmdharalds Since the vip-ci-bot already does php -l, should anything change there once this commit is used by the bot, to avoid duplicate reporting?

Yep, we will have to skip this sniff. There is also another sniff to ignore, so I will add an option to `vip-go-ci` to make sure we can skip custom sniffs.

**Comment by gudmdharalds** [comment]
This is the feature: https://github.com/Automattic/vip-go-ci/pull/31 -- aiming to deploy this after the weekend.

**Comment by nickdaugherty** [comment]
@gudmdharalds is Automattic/vip-go-ci#31 a blocker to merging this?

**Comment by gudmdharalds** [comment]
> @gudmdharalds is Automattic/vip-go-ci#31 a blocker to merging this?

Nope. Feel free to merge it.

**Pull request merged**